### PR TITLE
change copyright handling approach

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Drop our custom scanning of node modules (which broke during the Webpack 5 upgrade: nobody noticed) and replace it with the more standard Webpack approach of license files.

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Inspecting the tarball, I notice that it's now *smaller*.  This isn't doing exactly what I thought it was: the comments were still being included before, but as part of the minified .js output, instead of in separate files.

At the very least I need to edit a commit message or two here...

**Review by martinpitt — COMMENTED** [review]
This looks great, thanks! Running out of time today, note for myself: Check the generated LICENSES file and diff generated d/copyright against version from 244.

File: tools/build-debian-copyright [hunk 1]
```diff
@@ -111,6 +111,11 @@ for directory, subdirs, files in os.walk('dist'):
 
 for pattern in set.union(set(license_patterns), set(copyright_patterns)):
     if pattern not in used_patterns:
+        # We'll have no LICENSE.txt files in that dev builds
+        # so of course we won't use any of the patterns
+        if os.getenv('NODE_ENV') == 'development':
```
[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply tools/build-debian-copyright:116]: I didn't like how the indentation looked 😂

I also figured it kinda makes sense to do the check at the last possible moment before we'd bail out anyway in the name of "efficiency" on normal builds.

Those are my two Very Good Reasons for doing it this way :)

File: tools/build-debian-copyright [hunk 2]
```diff
@@ -144,29 +81,37 @@ with open(template_file, encoding='UTF-8') as f:
...
-    paragraph = f'Copyright:{copyrights}\nLicense: {license}'
-    # collect all modules with identical licence/copyright
-    if paragraph not in paragraphs:
-        paragraphs[paragraph] = set()
-    paragraphs[paragraph].update(module_users[moddir])
+
+for directory, subdirs, files in os.walk('dist'):
+    for file in files:
+        if not file.endswith('.LICENSE.txt'):
+            continue
```
[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply tools/build-debian-copyright:90]: ...and indeed we rely on that for the development mode case.

The thing that would catch the problem is the check for the unused patterns.

**Comment by martinpitt** [comment]
- Tarball file diff: OK, just added lots of *.LICENSE.txt and dropped tools/build-copying
- Binary debdiff clean (apart from dbgsym/version number noise)
- However, the generated tools/debian/copyright is incomplete. Most stuff that we had in [previous versions](https://salsa.debian.org/utopia-team/cockpit/-/blob/master/debian/copyright) was just for development dependencies (these must have slipped in when we moved to the current code of scanning node_modules -- that was not the case when we shipped a subset of `node_modules/` with just the runtime dependencies). But for example, "Copyright Dave Gandy 2016. All rights reserved." from font-awesome is missing, and we use that all over the place.
- Syntactically, the "File:" stanzas need to be complete globs, i. e. "Files: dist/apps/*", not just "dist/apps".
- Also, "File:" globs are not supposed to repeat themselves -- we need to sort these "inside out" to list all included copyrights and licenses for each dist/* webpack; right now the "primary key" is the copyright. This is already wrong in current main, and can be a follow-up.

**Review by martinpitt — APPROVED** [review]
As this is currently completely broken, I'm ok with landing as-is, but this definitively needs a follow-up.

**Comment by allisonkarlitskaya** [comment]
>     * Also, "File:" globs are not supposed to repeat themselves -- we need to sort these "inside out" to list all included copyrights and licenses for each dist/* webpack; right now the "primary key" is the copyright. This is already wrong in current main, and can be a follow-up.

[The spec](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#files-paragraph) seems to imagine exactly this scenario, and states:

> Since the license of the manual pages is the same as most other files in the package, the final paragraph above could instead be combined with the first paragraph, listing both copyright statements in one Copyright field. Whether to combine paragraphs with the same grant of license is left to the discretion of the author of the debian/copyright file.

**Comment by martinpitt** [comment]
@allisonkarlitskaya : Your cited manpage case is just an example of a specialization, see:

> Multiple Files paragraphs are allowed. The last paragraph that matches a particular file applies to it. More general paragraphs should therefore be given first, followed by more specific overrides

As we have multiple identical `File:` matches, only the last one applies and the others are formally ignored.

I'll look at cleaning this up, and will ask you for review.

**Comment by martinpitt** [comment]
We are also getting a warning in the build log:

    W: cockpit source: syntax-error-in-dep5-copyright debian/copyright: Cannot parse line 82: Facebook, Inc. and its affiliates.

This is due to wrong line continuation/duplication:

    Copyright: (c) Facebook, Inc. and its affiliates. 
    Facebook, Inc. and its affiliates.

**Comment by martinpitt** [comment]
Fixed in #16148

**Pull request merged**